src/test/java/com/epam/reportportal/cucumber/LaunchSystemAttributesTest.java

```diff
@@ -49,29 +50,39 @@ public static void initKeys() {
 	@Before
 	public void initLaunch() {
 		MockitoAnnotations.initMocks(this);
-		stepReporter.RP = Suppliers.memoize(() -> {
-
-			when(reportPortal.getParameters()).thenReturn(new ListenerParameters());
-			ListenerParameters parameters = reportPortal.getParameters();
-
-			StartLaunchRQ rq = new StartLaunchRQ();
-			rq.setAttributes(parameters.getAttributes() == null ? new HashSet<>() : parameters.getAttributes());
-			rq.getAttributes().addAll(SystemAttributesExtractor.extract("agent.properties"));
-
-			when(reportPortal.newLaunch(any(StartLaunchRQ.class))).thenReturn(new TestLaunch(parameters, rq, launchId));
-			return reportPortal.newLaunch(rq);
-		});
+		when(listenerParameters.getEnable()).thenReturn(true);
+		when(listenerParameters.getBaseUrl()).thenReturn("http://example.com");
+		when(listenerParameters.getIoPoolSize()).thenReturn(10);
+		when(listenerParameters.getBatchLogsSize()).thenReturn(5);
+		stepReporter = new StepReporter() {
+			@Override
+			protected ReportPortal buildReportPortal() {
+				return ReportPortal.create(reportPortalClient, listenerParameters);
+			}
+		};
 	}
 
 	@Test
 	public void shouldRetrieveSystemAttributes() {
-		TestLaunch testLaunch = (TestLaunch) stepReporter.RP.get();
-		StartLaunchRQ startLaunchRequest = testLaunch.getStartLaunchRQ();
+		final StartLaunchRQ[] startLaunchRequest = new StartLaunchRQ[1];
+		when(reportPortalClient.startLaunch(any(StartLaunchRQ.class))).then(t -> {
+			startLaunchRequest[0] = t.getArgumentAt(0, StartLaunchRQ.class);
```

Comment:
Well, there is Mockito ArgumentCaptor for that, check out here: https://github.com/reportportal/client-java/blob/develop/src/test/java/com/epam/reportportal/service/ItemLoggingContextMultiThreadTest.java#L151-L152

Please use straight way.

Reply:
fixed